cmd/run.go

```diff
@@ -435,19 +438,438 @@ func init() {
 	RootCmd.AddCommand(RunCmd)
 }
 
+func executeRun(runFilePath string, apps []runfileconfig.App) (bool, error) {
+	var exitWithError bool
+
+	// setup shutdown notify channel.
+	sigCh := make(chan os.Signal, 1)
+	setupShutdownNotify(sigCh)
+
+	runStates := make([]*runExec.RunExec, 0, len(apps))
+	for _, app := range apps {
+		// Set defaults if zero value provided in config yaml.
+		app.RunConfig.SetDefaultFromSchema()
+
+		// Validate validates the configs and modifies the ports to free ports, appId etc.
+		err := app.RunConfig.Validate()
+		if err != nil {
+			print.FailureStatusEvent(os.Stderr, "Error validating run config for app %s present in %s: %s", app.RunConfig.AppID, runFilePath, err.Error())
+			exitWithError = true
+			break
+		}
+
+		// Get Run Config for different apps.
+		runConfig := app.RunConfig
+
+		appLogWriter, err := app.SetAndGetAppLogWriter()
+		if err != nil {
+			print.StatusEvent(os.Stderr, print.LogFailure, "Error getting log file for app %s present in %s: %s", runConfig.AppID, runFilePath, err.Error())
+			exitWithError = true
+			break
+		}
+		daprdLogWriter, err := app.SetAndGetDaprdLogWriter()
+		if err != nil {
+			print.StatusEvent(os.Stderr, print.LogFailure, "Error getting log file for app %s present in %s: %s", runConfig.AppID, runFilePath, err.Error())
+			exitWithError = true
+			break
+		}
+		// create a multiwriter to write to both app and daprd log files.
+		appDaprdWriter := io.MultiWriter(appLogWriter, daprdLogWriter)
+		runState, err := startDaprdAndAppProcesses(&runConfig, app.AppDirPath, sigCh, daprdLogWriter, daprdLogWriter, appLogWriter, appLogWriter)
+		if err != nil {
+			print.StatusEvent(os.Stdout, print.LogFailure, "Error starting Dapr and app (%q): %s", app.AppID, err.Error())
+			print.StatusEvent(appDaprdWriter, print.LogFailure, "Error starting Dapr and app (%q): %s", app.AppID, err.Error())
+			exitWithError = true
```

Comment:
I'm not sure users will want to automatically shutdown *everything* in the event that *one* applications fails to start.  Or, at least, not *immediately*.  It might be good to have a user prompt in that case to ask if all should be shutdown and/or perhaps a command line switch to ignore failures.

My thinking is that not all applications must be running in order to test a given scenario; the fact that one application is misbehaving during startup shouldn't necessarily prevent the user from testing a scenario that does not involve that application.  As this isn't a production scenario, sometimes a "best effort" approach offers users the most flexibility.

Reply:
For such scenarios, can we allow users to explicitly specify a flag `--skip-shutdown-on-error` with `dapr run -f`, which will keep other applications running even if an application shuts down?

Reply:
I think that's reasonable; it'll fail the first time but, if users really want/need to run anyway, they'll run again with the flag.

A future improvement would then be to introduce the idea of cycling applications that fail, with the idea that errors may just be transient.  That could then lead further into "watch" like behavior that automatically cycle applications when source changes are detected.

Reply:
The "watch" like behavior was initially proposed in the dapr/proposals#6 proposal.
But then based on the discussion in that PR, any `orchestrator-like` or `watch-like` feature was removed from the proposal.

Reply:
Resolving this in-lieu of continuing the conversation in #1173

Reply:
Sounds good.